app/models/inning.rb

```diff
@@ -1,9 +1,18 @@
 class Inning < ApplicationRecord
   validates_presence_of :name, :start_date
+  validate :date_within_allowed_range
 
   has_many :turing_modules, dependent: :destroy
   has_many :students, through: :turing_modules
 
+  def date_within_allowed_range
+    if Inning.none?
+      # Execute as normal without triggering validation error
+    else current_inning_date = Inning.where(current: true).first.start_date
```

Comment:
A little syntax tip: `.where.first` can be replaced by `find_by`

Reply:
Thank you, switched this to use 'find_by_current(true)' to improve readability and syntax